apis/v1/httproute_types.go

```diff
@@ -119,6 +119,7 @@ type HTTPRouteSpec struct {
 	// +optional
 	// +kubebuilder:validation:MaxItems=16
 	// +kubebuilder:default={{matches: {{path: {type: "PathPrefix", value: "/"}}}}}
+	// +kubebuilder:validation:XValidation:message="While 16 rules and 64 matches are allowed, the total matches must be less than 256",rule="self.map(r, r.matches.size()).sum() < 256"
```

Comment:
This is still a 2x increase over the current upper limits, would prefer to keep the upper limit at 128 unless we have a very clear need for more within a single route. Would also be great to have some basic test cases to ensure this validation is working as intended.

Reply:
Ok, dropped to 128 and added a positive and negative test